pypdf/_page.py

```diff
@@ -612,21 +625,35 @@ def compute_unique_key(base_key: str) -> Tuple[str, bool]:
                 idx += 1
             return computed_key, False
 
+        if new_res1:
+            new_res = DictionaryObject()
+            new_res.update(res1.get(resource, DictionaryObject()).get_object())
+        else:
+            new_res = cast(DictionaryObject, res1[resource])
         page2res = cast(
             DictionaryObject, res2.get(resource, DictionaryObject()).get_object()
         )
         rename_res = {}
-        for key in sorted(page2res.keys()):
+        for key in page2res.keys():
```

Comment:
I think the sorting here was done one purpose to ensure the output is deterministic. See https://github.com/py-pdf/pypdf/pull/1543

Reply:
the sorting is performed further : L653-L656